### ISSUE TITLE
Training information for Nuplan

### ISSUE THREAD

**Issue by lan-lc** [Kguo-cs/ccil#1]
1. How do you split the training data into train/val/test? I saw NuPlan have six Las Vegas dataset. Do you use all of them?
2. Can you provide the following training details of NuPlan?  Epoch numbers, Final loss, # of GPU, Training Time, and early stopping.
3. Can you provide some checkpoints or pretrained model for NuPlan and provide the information of them? 

Thank you so much.

**Comment by Kguo-cs** [comment]
1. use all Las Vegas scenarios in train/val/test sets from the offiicial website (not just training data)
2. epoch 30, GPU: 3090*8, find the model with minimum collision rate
3. I have uploaded one checkpoint.

**Comment by Kguo-cs** [comment]
3. I have uploaded one checkpoint in the google drive.

**Comment by lan-lc** [comment]
Thank you so much for your rapid reply and for uploading one checkpoint! 
Can you also share the final loss value and the total training time for the second point? Thank you.

**Comment by Kguo-cs** [comment]
Sorry. The loss term is not important so I do not remember it. The total training time should be about three days.

**Comment by lan-lc** [comment]
No worries!

I'm inquiring about the loss value because I'm currently training on NuPlan's mini dataset and want to check for potential overfitting due to its significantly smaller size.

A new concern for me is the size difference between the pretrained model "nuplan.ckpt" you uploaded (70 MB) and the checkpoint I trained, labeled "'epoch=01-val_loss=0.0000.ckpt" (18 MB). Is there anything specific I should modify to achieve a similar size? Additionally, could you explain why the "val_loss" is reported as 0.0000? Furthermore, I'd appreciate it if you could clarify the other models present under the pretrained folder, like "planning_model_20201208_*.pt." models. 

Thanks in advance for your help!"

**Comment by Kguo-cs** [comment]
Sorry. The checkpoint I uploaded is with 256 hiddden size. I have uploaded a checkpoint with hidden size of 128.

**Comment by Kguo-cs** [comment]
Val loss is not important so I does not stored it during training. The detail of the other models can be found in the appendix of our paper.

**Comment by lan-lc** [comment]
It seems like the current code (google drive) can not load the 128 checkpoint. (different layer name) 

On the other hand, when I change the config/nuplan/nuplan.yaml's d_model from 128 to 256, eval.py can load the model. 
However, the results is different from the paper, do you know which part should I also need to change? Thank you!

───────────────────────────────────────────────────────
        Test metric               DataLoader 0
───────────────────────────────────────────────────────
       val/acc_mean             1.006670355796814
    val/collision_front                0.0
    val/collision_rate         0.3411552309989929
    val/collision_rear                 0.0
    val/collision_side                 0.0
        val/comfort            0.06705630570650101
 val/displacement_error_l2     22.631425857543945
val/distance_ref_trajectory     9.232389450073242
       val/jerk_mean            3.707332134246826
        val/off_ref            0.4783393442630768
       val/off_road            0.2779783308506012
      val/x_diff_mean          0.7205662727355957
      val/y_diff_mean          0.15002186596393585
     val/yaw_diff_mean        0.011783784255385399

**Comment by lan-lc** [comment]
Or maybe you can provide the config that you used to train NuPlan d256?
Thank you in advance.

**Comment by Kguo-cs** [comment]
Did you process the official test dataset as I do?

**Comment by Kguo-cs** [comment]
'yaw_w' = {int} 1
'acc_w' = {float} 0.1
'control_w' = {int} 1
'lqr_iter' = {int} 10
'yaw_vel_w' = {int} 1
'yaw_acc_w' = {float} 0.1
'yaw_control_w' = {float} 1e-06

**Comment by Kguo-cs** [comment]
I have uploaded our used test dataset in the google drive (data/nuplan_meta)

**Comment by Kguo-cs** [comment]
Sorry, I try to upload our used test dataset in the google drive but it is too large (21.2G). So I upload in the onedrive (https://connecthkuhk-my.sharepoint.com/:f:/g/personal/u3006612_connect_hku_hk/EtAxqEfFQzJChAwf0KI9u5AB3_qtLQkLJVuwRpBSCb0XFw?e=8sou76)

**Comment by lan-lc** [comment]
Sorry for keep asking questions and thank you for answering. 

I simply use the test database provided by the official webpage. I have checked that it is the same as yours (./data/nuplan_meta/nuplan.yaml).

Even changed the config file (./config/nuplan/nuplan.yaml) according to the one you just posted the evaluation results are almost the same. 
────────────────────────────────────────────────────
        Test metric               DataLoader 0
────────────────────────────────────────────────────
       val/acc_mean            0.8341479897499084
    val/collision_front                0.0
    val/collision_rate         0.35379061102867126
    val/collision_rear                 0.0
    val/collision_side                 0.0
        val/comfort           0.043117500841617584
 val/displacement_error_l2     22.281099319458008
val/distance_ref_trajectory     8.823013305664062
       val/jerk_mean            2.186784267425537
        val/off_ref            0.4783393442630768
       val/off_road            0.2797833979129791
      val/x_diff_mean          0.7225600481033325
      val/y_diff_mean          0.14154843986034393
     val/yaw_diff_mean        0.012174375355243683

My current config for evaluation is:

model_params:
    history_num_frames_agents: 2
    future_num_frames: 15

    step_time: 0.1

    dropout: 0.1
    head_num: 8
    d_model: 256

    local_num_layers: 3
    global_num_layers: 6
    causal_num_layers: 3
    causal_len: 15
    causal_interval: 2

    lane_feat_dim: 13
    cross_feat_dim: 6
    ego_feat_dim: 2
    agent_feat_dim: 10
    act_dim: 3

    prev_weight: 0.3

    read_into_memory: True

    ignore_ego: True

    yaw_w: 1

    acc_w: 0.1
    control_w: 1
    lqr_iter: 10

    yaw_vel_w: 1
    yaw_acc_w: 0.1
    yaw_control_w: 0.000001
eval:
  num_scenes_to_unroll: 1024
  batch_size: 64

**Comment by Kguo-cs** [comment]
We use the nuplan v1.0 data ( recent is v1.1). I'm uploading the test data we used in onedrive (https://connecthkuhk-my.sharepoint.com/:f:/g/personal/u3006612_connect_hku_hk/EtAxqEfFQzJChAwf0KI9u5AB3_qtLQkLJVuwRpBSCb0XFw?e=yQVYr5).

**Comment by lan-lc** [comment]
Hi, when i replaced the ./data/nuplan_meta/test.npz with your test.npz, the nuplan_dataset doesn't not read your file and start to generate a new test.npz with the nuplan dataset we have.  Do you know how to fix it? Thank you. 

Follow up:
It seems like current nuplan scenario is different from the old version. Hence, it cannot read your scenario properly.

**Comment by Kguo-cs** [comment]
remove the "self.scenarios=data["scenarios"][::self.sim_len]" code at nuplan_dataset due to the data path difference .

        try:
            data=np.load('./data/nuplan_meta/'+type+".npz",allow_pickle=True)

           # self.scenarios=data["scenarios"][::self.sim_len]

**Comment by lan-lc** [comment]
Thank you for the guidance! It worked! The results show that the pretrained 128 model has a 7.9% collision rate on your shared test dataset. However, there's an off-road rate of around 11% during the same evaluation. Is this expected? Additionally, when I train on NuPlan 1.1, the collision rate consistently exceeds 13%. Any recommendations?
Thank you again.

**Comment by Kguo-cs** [comment]
What's your LQR parameter for my pretrained 128(sure?) model? (I think you can only load 256 model) . How do you train your only model?
